java/vector/src/test/java/org/apache/arrow/vector/TestMapVector.java

```diff
@@ -624,6 +625,423 @@ public void testMapWithListValue() throws Exception {
     }
   }
 
+  @Test
+  public void testMapWithMapValue() throws Exception {
+    try (MapVector mapVector = MapVector.empty("sourceVector", allocator, false)) {
+
+      UnionMapWriter mapWriter = mapVector.getWriter();
+      MapWriter valueWriter;
+
+      // we are essentially writing Map<Long, Map<Long, Long>>
```

Comment:
What would help (especially the people who will edit this test in the future) is to have a description in comments of how the map will look like after you have added whatever data you want to use for test. I had used this in several places in TestValueVector, TestListVector etc and it helps a lot with visualizing and debugging.

Reply:
I added a comment showing the resulting Map that the code is creating. I tried to pick a syntax that was both concise yet clear, however it gets challenging as the data type gets more complex.